hera_mc/cm_utils.py

```diff
@@ -126,47 +121,82 @@ def add_date_time_args(parser):
+        d = "{:%Y-%m-%d %H:%M:%S}".format(d.datetime)
+    return d
+
+
 def get_astropytime(_date, _time=0):
     """
-    Take in various incarnations of _date/_time and return an astropy.Time object
+    Take in various incarnations of _date/_time and return an astropy.Time object or None.
+    - _time should either be a number in hours or HH:MM[:SS]
     """
 
-    add_time = 0.
-    return_date = None
     if isinstance(_date, Time):
-        return_date = _date
-    elif _date is None or _date is False:
-        return_date = _date
-    else:
+        return _date
+    if isinstance(_date, datetime.datetime):
+        return Time(_date, format='datetime')
+    if _date is None or _date is False:
+        return None
+    if isinstance(_date, (int, long, float)):
+        if int(_date) > 1000000000:
+            return Time(_date, format='gps')
+        raise ValueError('Invalid format:  date as a number should be gps time, not {}.'.format(_date))
+    if isinstance(_date, str):
         if _date == '<':
-            return_date = Time(PAST_DATE, scale='utc')
-        elif _date == '>':
-            return_date = future_date()
-        elif _date.lower().replace('/', '') == 'na' or _date.lower() == 'none':
-            return_date = None
-        elif _date.lower() == 'now':
-            return_date = Time.now()
-        elif _date is not None:
-            _date = _date.replace('/', '-')
-            try:
-                return_date = Time(_date, scale='utc')
-            except ValueError:
-                raise ValueError('Invalid format:  date should be YYYY/M/D or YYYY-M-D')
-            s_time = str(_time)
-            if ':' in s_time:
-                data = s_time.split(':')
-                add_time = float(data[0]) * 3600.0 + float(data[1]) * 60.0
-                if len(data) == 3:
-                    add_time += float(data[2])
-            else:
-                try:
-                    add_time = float(_time) * 3600.0
-                except ValueError:
-                    raise ValueError('Invalid format:  time should be H[:M[:S]] '
-                                     '(HMS can be float or int)')
-        if return_date is not None:
-            return_date += TimeDelta(add_time, format='sec')
-    return return_date
+            return Time(PAST_DATE, scale='utc')
+        if _date == '>':
+            return future_date()
+        if _date.lower() == 'now' or _date.lower() == 'current':
+            return Time.now()
+        if _date.lower().replace('/', '') == 'na' or _date.lower() == 'none':
+            return None
+        _date = _date.replace('/', '-')
+        try:
+            return_date = Time(_date, scale='utc')
+        except ValueError:
+            raise ValueError('Invalid format:  date should be YYYY/M/D or YYYY-M-D')
+        if isinstance(_time, (float, int)):
+            return return_date + TimeDelta(_time * 3600.0, format='sec')
+        if isinstance(_time, str) and ':' in _time:
```

Comment:
So, if this function is called to parse a command line date/time argument, the `_time` argument will be passed to this function as a `str` object. If the user gave the time as just an integer (eg "11"), this function will raise a ValueError since this branch only gets triggered if the `_time` string contains a ":" as well. It seems like it would be helpful for it too handle the no-colon case too, especially since the case numeric inputs is handled OK.

Also note that if the time is given as "11:00:59:ABC", the code will silently handle it as if the user gave "11:00". I generally think parsing should be conservative about these kinds of cases since a mistake in this sort of input could easily signify that something bigger has gone wrong in whatever code is driving this routine.